ci/deps/actions-38-downstream_compat.yaml

```diff
@@ -1,54 +1,66 @@
+# Non-dependencies that pandas utilizes or has compatibility with pandas objects
 name: pandas-dev
 channels:
   - conda-forge
 dependencies:
   - python=3.8
   - pip
 
-  # tools
+  # test dependencies
   - cython>=0.29.24
   - pytest>=6.0
   - pytest-xdist>=1.31
   - hypothesis>=5.5.3
   - pytest-cov>=2.10.1  # this is only needed in the coverage build, ref: GH 35737
+  - nomkl
 
-  # pandas dependencies
-  - aiobotocore<2.0.0  # GH#44311 pinned to fix docbuild
+  # required dependencies
+  - numpy
+  - python-dateutil
+  - pytz
+
+  # optional dependencies
   - beautifulsoup4
-  - boto3
-  - botocore>=1.11
-  - dask
+  - blosc
   - fastparquet>=0.4.0
   - fsspec>=0.7.4
-  - gcsfs>=0.6.0
-  - geopandas
+  - gcsfs
   - html5lib
+  - jinja2
+  - lxml
   - matplotlib
-  - nomkl
   - numexpr
-  - numpy=1.18
   - odfpy
   - openpyxl
   - pandas-gbq
   - psycopg2
   - pyarrow>=1.0.1
   - pymysql
   - pytables
-  - python-snappy
-  - python-dateutil
-  - pytz
+  - pyxlsb
   - s3fs>=0.4.0
-  - scikit-learn
   - scipy
   - sqlalchemy
-  - statsmodels
   - xarray
   - xlrd
   - xlsxwriter
   - xlwt
+
+  # downstream packages
+  - aiobotocore<2.0.0  # GH#44311 pinned to fix docbuild
+  - boto3
+  - botocore>=1.11
+  - dask
+  - ipython
+  - geopandas
+  - python-snappy
+  - seaborn
+  - scikit-learn
+  - statsmodels
   - brotlipy
   - coverage
   - pandas-datareader
-  - pyxlsb
+  - pyyaml
   - pip:
+    - py
```

Comment:
Why add ``py``? Also I'm pretty sure this is available on conda-forge. https://anaconda.org/conda-forge/py

Reply:
Ah cool I'll install the conda-forge version.

We have some tests that run on `py.path` objects

https://github.com/pandas-dev/pandas/blob/cbefe18400e5e3bfd51ea5e979973a57d6803319/pandas/tests/io/test_common.py#L42

Thought maybe in a follow up we should remove/discourage because it appears py.path is in maintenance only mode.